generate_release_notes.py

```diff
@@ -125,173 +149,241 @@ def get_repo_list(github_org, github):
+                                issue_titles_other.extend(github_issues)
 
-                            # Fallback to PR description and title at the end if no Issues are found.
                             else:
-                                pull_requests_missing_epics.add(pr.title.strip() + f" (Pull Request [#{pr.number}]({pr.html_url}))")
-                        except Exception as e:
-                            logging.error(f"Error processing PR #{pr.number} for repo {repo.name}: {e}")
+                                pull_requests_missing_epics.add(f"{pr.title.strip()} (Pull Request [#{pr.number}]({pr.html_url}))")
+                        except GithubException as error:
+                            logging.error("Error processing PR #%d for repo %s: %s", pr.number, repo.name, error)
                             skipped_prs.append(f"PR #{pr.number} in repo {repo.name} failed to process")
-                else:
-                    logging.warning(f"No pull requests found for repo {repo.name}")
-                # Generate release notes for repo
-                release_notes += get_release_notes(repo.name, args.version, issue_titles_bugs, issue_titles_epics, issue_titles_other, commit_only, pull_requests_missing_epics)
-                logging.info('Generated release note for repo: ' + github_repo)
-        # write skipped repositories and PRs to the release notes
+
+                notes += get_release_notes(
+                    repo.name, parsed_args.version, issue_titles_epics,
+                    issue_titles_other, commit_only, pull_requests_missing_epics, pr_mapping
+                )
+                logging.info("Generated release note for repo: %s", github_repo)
+
         if skipped_repos:
-            release_notes += "\n\n### Skipped Repositories\n"
-            release_notes += "\n".join(skipped_repos)
+            notes += "\n\n### Skipped Repositories\n" + "\n".join(skipped_repos)
+
         if skipped_prs:
-            release_notes += "\n\n### Skipped Pull Requests\n"
-            release_notes += "\n".join(skipped_prs)
-        # Write release notes to file
-        pathlib.Path(args.output_file).unlink(missing_ok=True)
-        with open(args.output_file, "w") as f:
-            f.write(release_notes)
-
-    except Exception as e:
-        logging.error(e)
-        sys.exit(1)
-    except KeyboardInterrupt:
-        logging.error("Keyboard interrupt")
+            notes += "\n\n### Skipped Pull Requests\n" + "\n".join(skipped_prs)
+
+        pathlib.Path(parsed_args.output_file).unlink(missing_ok=True)
+        with open(parsed_args.output_file, "w", encoding="utf-8") as file:
+            file.write(notes)
+
+    except GithubException as error:
+        logging.error("%s", error)
         sys.exit(1)
 
 if __name__ == "__main__":
     parser = argparse.ArgumentParser()
-    parser.add_argument("--github-token", required=True)  
+    parser.add_argument("--github-token", required=True)
```

Comment:
Please add descriptions for these arguments and default values where it makes sense. Like stable branch or organizations or jira URL since those will likely always be the same.

Reply:
Added default values and descriptions